src/VirtualClient/VirtualClient.Actions/MongoDB/MongoDBMetricsParser.cs

```diff
@@ -0,0 +1,160 @@
+﻿namespace VirtualClient.Actions
+{
+    using System;
+    using System.Collections.Generic;
+    using System.Linq;
+    using System.Text.Json;
+    using System.Text.RegularExpressions;
+    using VirtualClient.Contracts;
+
+    /// <summary>
+    /// Parser for MongoDB benchmark output.
+    /// </summary>
+    public class MongoDBMetricsParser : MetricsParser
+    {
+        /// <summary>
+        /// Sectionize by one or more empty lines.
+        /// </summary>
+        private static readonly Regex MongoDBSectionDelimiter = new (@"(\n)(\s)*(\n)", RegexOptions.ExplicitCapture);
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="MongoDBMetricsParser"/> class.
+        /// </summary>
+        /// <param name="rawText">Raw text which is output of the MongoDB workload</param>
+        public MongoDBMetricsParser(string rawText)
+            : base(rawText)
+        {
+        }
+
+        /// <summary>
+        /// Logic to parse and read metrics8
+        /// </summary>
+        public override IList<Metric> Parse()
+        {
+            try
+            {
+                this.Preprocess();
+                Dictionary<string, Tuple<string, double>> metricsMap = GetMetricsMap(this.Sections["Metrics"]);
+                List<Metric> metrics = new List<Metric>();
+                var metricRelativity = MetricRelativity.Undefined;
+
+                foreach (var entry in metricsMap)
+                {
+                    if (entry.Key == "OVERALL-Throughput")
+                    {
+                        metricRelativity = MetricRelativity.HigherIsBetter;
+                    }
+                    else if (entry.Key.Contains("AverageLatency"))
+                    {
+                        metricRelativity = MetricRelativity.LowerIsBetter;
+                    }
+                    else if (entry.Key.Contains("95thPercentileLatency"))
+                    {
+                        metricRelativity = MetricRelativity.LowerIsBetter;
+                    }
+
+                    metrics.Add(new Metric(entry.Key, entry.Value.Item2, entry.Value.Item1, metricRelativity));
+
+                }
+
+                return metrics;
+            }
+            catch (JsonException exc)
+            {
+                throw new WorkloadResultsException("Workload results parsing failure. The example workload results are not valid or are not formatted in a valid JSON.", exc, ErrorReason.WorkloadResultsParsingFailed);
+            }
+        }
+
+        /// <summary>
+        /// Logic for preprocessing
+        /// </summary>
+        protected override void Preprocess()
+        {
+            RegexOptions options = RegexOptions.None;
+            var regex = new Regex("[ ]{2,}", options);
+            this.PreprocessedText = regex.Replace(this.RawText, " ");
+            string pattern = @"(?=\[OVERALL\])"; // Pattern to find the position before the word "[OVERALL]"
```

Comment:
Rather than having comment update the string name so that it's self readable and also update the rgx name below created from the pattern string